lhotse/recipes/librilight.py

```diff
@@ -0,0 +1,158 @@
+"""
+About the librilight corpus
+
+Libri-light is a benchmark for the training of automatic speech recognition (ASR)
+systems with limited or no supervision.
+
+It contains a large dataset of 60K hours of unlabelled speech from audiobooks in 
+English and a small labelled dataset (10h, 1h, and 10 min) plus metrics,
+trainable baseline models, and pretrained models that use these datasets.
+
+It is covered in more detail at https://arxiv.org/abs/1912.07875.
+
+This data is very huge - please download manually at LIBRILIGHT_URL.
+"""
+
+import logging
+import os
+from collections import defaultdict
+from concurrent.futures.thread import ThreadPoolExecutor
+from pathlib import Path
+from typing import Dict, List, Optional, Sequence, Tuple, Union
+
+from tqdm.auto import tqdm
+
+from lhotse.audio import Recording, RecordingSet
+from lhotse.recipes.utils import manifests_exist
+from lhotse.supervision import SupervisionSegment, SupervisionSet
+from lhotse.utils import Pathlike
+
+LIBRILIGHT = ("small", "medium", "large")
+
+LIBRILIGHT_URL = (
+    "https://dl.fbaipublicfiles.com/librilight/data/small.tar",
+    "https://dl.fbaipublicfiles.com/librilight/data/medium.tar",
+    "https://dl.fbaipublicfiles.com/librilight/data/large.tar",
+)
+
+
+def _parse_utterance(
+    corpus_dir: Pathlike,
+    audio_path: Pathlike,
+) -> Optional[Tuple[Recording, SupervisionSegment]]:
+    file_name = audio_path.replace(".flac", "").replace(str(corpus_dir) + "/", "")
+    speaker = audio_path.split("/")[-3]
+    audio_path = Path(audio_path).resolve()
+
+    if not audio_path.is_file():
+        logging.warning(f"No such file: {audio_path}")
+        return None
+
+    recording = Recording.from_file(
+        path=audio_path,
+        recording_id=file_name,
+    )
+    segment = SupervisionSegment(
+        id=file_name,
+        recording_id=file_name,
+        start=0.0,
+        duration=recording.duration,
+        channel=0,
+        language="English",
+        speaker=speaker,
+    )
+
+    return recording, segment
+
+
+def _prepare_subset(
+    subset: str,
+    corpus_dir: Pathlike,
+    num_jobs: int = 1,
+) -> Tuple[RecordingSet, SupervisionSet]:
+    """
+    Returns the RecodingSet and SupervisionSet given a dataset part.
+    :param subset: str, the name of the subset.
+    :param corpus_dir: Pathlike, the path of the data dir.
+    :return: the RecodingSet and SupervisionSet for train and valid.
+    """
+    corpus_dir = Path(corpus_dir)
+    part_path = corpus_dir / subset
+    audio_paths = []
+    for root, dirs, files in os.walk(part_path):
```

Comment:
Can this loop be turned into a one-liner with `Path(part_path).rglob("*.flac")`?

Reply:
Sure.